src/api/dto/from/dtoToCriteria.ts

```diff
@@ -0,0 +1,152 @@
+import { FeeFineTypeDTO } from '../../queries/useFeeFineTypes';
+import { LocationDTO } from '../../queries/useLocations';
+import {
+  BursarExportFilterCondition,
+  BursarExportFilterDTO,
+  BursarExportFilterNegation,
+} from '../types';
+
+// inverse of ../to/criteriaToFilterDto
+export default function dtoToCriteria(
+  filter: BursarExportFilterDTO,
+  feeFineTypes: FeeFineTypeDTO[],
+  locations: LocationDTO[]
+): CriteriaGroup | CriteriaTerminal {
+  switch (filter.type) {
+    case 'Age':
+      return {
+        type: CriteriaTerminalType.AGE,
+        operator: filter.condition as ComparisonOperator,
+        numDays: filter.numDays.toString(),
+      };
+    case 'Amount':
+      return {
+        type: CriteriaTerminalType.AMOUNT,
+        operator: filter.condition as ComparisonOperator,
+        amountDollars: (filter.amount / 100).toFixed(2),
+      };
+    case 'FeeFineOwner':
+      return {
+        type: CriteriaTerminalType.FEE_FINE_OWNER,
+        feeFineOwnerId: filter.feeFineOwner,
+      };
+    case 'FeeType':
+      return {
+        type: CriteriaTerminalType.FEE_FINE_TYPE,
+        feeFineTypeId: filter.feeFineTypeId,
+        feeFineOwnerId: getFeeFineOwnerId(filter.feeFineTypeId, feeFineTypes),
+      };
+    case 'Location':
+      return {
+        type: CriteriaTerminalType.LOCATION,
+        locationId: filter.locationId,
+        ...getLocationProperties(filter.locationId, locations),
+      };
+    case 'PatronGroup':
+      return {
+        type: CriteriaTerminalType.PATRON_GROUP,
+        patronGroupId: filter.patronGroupId,
+      };
+    case 'ServicePoint':
+      return {
+        type: CriteriaTerminalType.SERVICE_POINT,
+        servicePointId: filter.servicePointId,
+      };
+
+    case 'Condition':
+      return dtoToConditionCriteria(filter, feeFineTypes, locations);
+
+    case 'Negation':
+      return dtoToNegationCriteria(filter, feeFineTypes, locations);
+
+    case 'Pass':
+    default:
+      return {
+        type: CriteriaTerminalType.PASS,
+      };
+  }
+}
+
+export function dtoToConditionCriteria(
+  filter: BursarExportFilterCondition,
+  feeFineTypes: FeeFineTypeDTO[],
+  locations: LocationDTO[]
+): CriteriaGroup {
+  if (filter.operation === 'AND') {
+    return {
+      type: CriteriaGroupType.ALL_OF,
+      criteria: filter.criteria.map((criteria) =>
+        dtoToCriteria(criteria, feeFineTypes, locations)
+      ),
+    };
+  } else {
+    return {
+      type: CriteriaGroupType.ANY_OF,
+      criteria: filter.criteria.map((criteria) =>
+        dtoToCriteria(criteria, feeFineTypes, locations)
+      ),
+    };
+  }
+}
+
+export function dtoToNegationCriteria(
+  filter: BursarExportFilterNegation,
+  feeFineTypes: FeeFineTypeDTO[],
+  locations: LocationDTO[]
+): CriteriaGroup {
+  // NOR gets displayed as "None of" in the UI, so we flatten the inner OR
+  if (
+    filter.criteria.type === 'Condition' &&
+    filter.criteria.operation === 'OR'
```

Comment:
Is this a viable way to do this? Or should I make an enum and put it in the `src/types` directory?

<img width="883" alt="Screenshot 2024-01-24 at 7 44 03 PM" src="https://github.com/folio-org/ui-plugin-bursar-export/assets/97154067/6a76f27c-54a3-4549-a35c-360138151ec7">

Reply:
We don't have any constants for `Condition` or `Negation`. I can add constants for them, but I don't know what I would define the enum as because they are the value of `type`. Or I can add them as direct constants via `export const CONDITION = 'Condition'`.